addons/survey/static/src/js/survey_result.js

```diff
@@ -1,42 +1,102 @@
-    return Promise.reject("DOM doesn't contain '.js_surveyresult'");
-}
-
-    // The given colors are the same as those used by D3
-    var D3_COLORS = ["#1f77b4","#ff7f0e","#aec7e8","#ffbb78","#2ca02c","#98df8a","#d62728",
-                        "#ff9896","#9467bd","#c5b0d5","#8c564b","#c49c94","#e377c2","#f7b6d2",
-                        "#7f7f7f","#c7c7c7","#bcbd22","#dbdb8d","#17becf","#9edae5"];
-
-    //Script For Pagination
-    var survey_pagination = $('.pagination');
-    $.each(survey_pagination, function(index, pagination){
-        var question_id = $(pagination).attr("data-question_id");
-        var limit = $(pagination).attr("data-record_limit"); //Number of Record Par Page. If you want to change number of record per page, change record_limit in pagination template.
-        $('#table_question_'+ question_id +' tbody tr:lt('+limit+')').removeClass('d-none');
-        $('#pagination_'+question_id+' li a').click(function(event){
-            event.preventDefault();
-            $('#pagination_'+question_id+' li').removeClass('active');
-            $(this).parent('li').addClass('active');
-            $('#table_question_'+ question_id +' tbody tr').addClass('d-none');
-            var num = $(this).text();
-            var min = (limit * (num-1))-1;
-            if (min == -1){
-                $('#table_question_'+ question_id +' tbody tr:lt('+ limit * num +')').removeClass('d-none');
+var ajax = require('web.ajax');
+var publicWidget = require('web.public.widget');
+
+// The given colors are the same as those used by D3
+var D3_COLORS = ["#1f77b4","#ff7f0e","#aec7e8","#ffbb78","#2ca02c","#98df8a","#d62728",
+                    "#ff9896","#9467bd","#c5b0d5","#8c564b","#c49c94","#e377c2","#f7b6d2",
+                    "#7f7f7f","#c7c7c7","#bcbd22","#dbdb8d","#17becf","#9edae5"];
+
+publicWidget.registry.SurveyResultWidget = publicWidget.Widget.extend({
+    selector: '.o_survey_result',
+    events: {
+        'click td.survey_answer i.fa-filter': '_onSurveyAnswerFilterClick',
+        'click .clear_survey_filter': '_onClearFilterClick',
+        'click span.filter-all': '_onFilterAllClick',
+        'click span.filter-finished': '_onFilterFinishedClick',
+    },
+
+    //--------------------------------------------------------------------------
+    // Widget
+    //--------------------------------------------------------------------------
+
+    /**
+    * @override
+    */
+    willStart: function () {
+        var url = '/web/webclient/locale/' + (document.documentElement.getAttribute('lang') || 'en_US').replace('-', '_');
+        var localeReady = ajax.loadJS(url);
+        return Promise.all([this._super.apply(this, arguments), localeReady]);
+    },
+
+    /**
+    * @override
+    */
+    start: function () {
+        var self = this;
+
+        //Script For Pagination
+        var survey_pagination = $('.pagination');
+        $.each(survey_pagination, function(index, pagination){
```

Comment:
Meh.

This could almost be in a `SurveyResultPaginationWidget` with nice and clean `events` and `trigger_up`...

Reply:
meh.

Reply:
done in next commit (by awa btw)